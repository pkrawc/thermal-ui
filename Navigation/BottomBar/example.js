import React, { Component } from 'react'
import { BottomBar } from '../BottomBar'
import { Button } from '../../General/Button'
import { Container } from '../../Layout'

class BottomBarExample extends Component {
  constructor(props) {
    super(props)
    this.state = { bbActive: false }
    this.toggleState = (e) => {
      e.preventDefault
      this.setState({bbActive: !this.state.bbActive})
    }
  }
  render() {
    return (
      <Container>
        <Button onClick={this.toggleState}>Click to toggle bottom bar</Button>
        <BottomBar active={this.state.bbActive}>Hello Bottom Bar</BottomBar>
      </Container>
    )
  }
}

export default {
  title: 'Bottom Bar',
  examples: [
    (
      <BottomBarExample />
    )
  ]
}
