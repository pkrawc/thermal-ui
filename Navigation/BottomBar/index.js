import React, { PropTypes } from 'react'
import styled from 'styled-components'
import { Motion, spring } from 'react-motion'
import { colors, shadows } from 'variables'

const BottomBarWrapper = styled.div`
  position: fixed;
  top: 0; right: 0; bottom: 0; left: 0;
  visibility: ${({active}) => active ? 'initial' : 'hidden'};
  pointer-events: ${({active}) => active ? 'initial' : 'none'};
  background-color: ${colors.disabled};
  transition: visibility 150ms ease-in-out;
`

const Bar = styled.div`
  margin: 0.5em;
  padding: 1em;
  position: fixed;
  right: 0; bottom: 0; left: 0;
  background-color: ${colors.light};
  border-radius: 2px;
  box-shadow: ${shadows.small};
  font-size: 1em;
  min-height: 2.5em;
  opacity: ${({active}) => active ? 1 : 0};
  transform: ${({y}) => `translateY(${y}%)`};
  transition: opacity 150ms ease-in-out;
  z-index: 999;
`

export function BottomBar({
  children,
  toggle,
  active
}) {
  return (
    <BottomBarWrapper active={active} onClick={toggle}>
      <Motion style={{y: spring(active ? 0 : 100, {stiffness: 250, damping: 30})}}>
        {({y}) =>
          <Bar
            active={active}
            y={y}
            onClick={(e) => {e.stopPropagation()}}>
            {children}
          </Bar>}
      </Motion>
    </BottomBarWrapper>
  )
}

BottomBar.propTypes = {
  active: PropTypes.bool.isRequired,
  toggle: PropTypes.func.isRequired
}
