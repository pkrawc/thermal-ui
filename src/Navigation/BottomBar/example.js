import React, { Component } from 'react'
import { BottomBar } from '../BottomBar'
import { Button } from '../../General/Button'
import { Container } from '../../Layout'

class BottomBarExample extends Component {
  constructor(props) {
    super(props)
    this.state = { bbActive: false }
    this.toggleState = _ => {
      this.setState({bbActive: !this.state.bbActive})
    }
  }
  render() {
    return (
      <Container>
        <Button onClick={this.toggleState}>Click to toggle bottom bar</Button>
        <BottomBar active={this.state.bbActive} toggle={this.toggleState}>
          <Button onClick={() => {console.log('hello bottom button')}}>Hello Bottom Bar Button</Button>
        </BottomBar>
      </Container>
    )
  }
}

export default {
  title: 'Bottom Bar',
  example: BottomBarExample
}
