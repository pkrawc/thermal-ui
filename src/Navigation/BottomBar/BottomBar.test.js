import React from 'react'
import renderer from 'react-test-renderer'
import { BottomBar } from '../BottomBar'

it('renders bottom bar correctly', () => {
  const tree = renderer.create(
    <BottomBar>
      Hello World
    </BottomBar>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})
