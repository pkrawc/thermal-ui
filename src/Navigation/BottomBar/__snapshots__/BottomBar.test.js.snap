exports[`test renders bottom bar correctly 1`] = `
<div
  className="hiqgSn"
  onClick={[Function]}>
  <div
    className="dwNke"
    onClick={[Function]}
    y={100}>
    Hello World
  </div>
</div>
`;
